pkg/kinflate/configmap_secret.go

```diff
@@ -18,74 +18,114 @@ package kinflate
 
 import (
 	"crypto/tls"
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/meta"
-	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
 	"k8s.io/kubectl/pkg/kinflate/configmapandsecret"
 	"k8s.io/kubectl/pkg/kinflate/hash"
+	kutil "k8s.io/kubectl/pkg/kinflate/util"
 )
 
-func populateMap(m map[groupVersionKindName]newNameObject, obj runtime.Object, newName string) error {
+func populateMap(m map[kutil.GroupVersionKindName]*unstructured.Unstructured, obj *unstructured.Unstructured, newName string) error {
 	accessor, err := meta.Accessor(obj)
 	if err != nil {
 		return err
 	}
 	oldName := accessor.GetName()
 	gvk := obj.GetObjectKind().GroupVersionKind()
-	gvkn := groupVersionKindName{
-		gvk:  gvk,
-		name: oldName,
-	}
+	gvkn := kutil.GroupVersionKindName{GVK: gvk, Name: oldName}
+
 	if _, found := m[gvkn]; found {
 		return fmt.Errorf("cannot use a duplicate name %q for %s", oldName, gvk)
 	}
 	accessor.SetName(newName)
-	m[gvkn] = newNameObject{
-		newName: newName,
-		obj:     obj,
-	}
+	m[gvkn] = obj
 	return nil
 }
 
-func populateMapOfConfigMapAndSecret(r *resource, m map[groupVersionKindName]newNameObject) error {
-	for _, cm := range r.configmaps {
-		corev1cm, err := constructConfigMap(cm)
+func populateConfigMapAndSecretMap(manifest *manifest.Manifest, m map[kutil.GroupVersionKindName]*unstructured.Unstructured) error {
+	for _, cm := range manifest.Configmaps {
+		unstructuredConfigMap, nameWithHash, err := constructConfigmapAndGenerateName(cm)
 		if err != nil {
 			return err
 		}
-		h, err := hash.ConfigMapHash(corev1cm)
-		if err != nil {
-			return err
-		}
-		nameWithHash := fmt.Sprintf("%s-%s", corev1cm.Name, h)
-		err = populateMap(m, corev1cm, nameWithHash)
+		err = populateMap(m, unstructuredConfigMap, nameWithHash)
 		if err != nil {
 			return err
 		}
 	}
 
-	for _, secret := range r.secrets {
-		corev1secret, err := constructSecret(secret)
-		if err != nil {
-			return err
-		}
-		h, err := hash.SecretHash(corev1secret)
+	for _, secret := range manifest.Secrets {
+		unstructuredSecret, nameWithHash, err := constructSecretAndGenerateName(secret)
 		if err != nil {
 			return err
 		}
-		nameWithHash := fmt.Sprintf("%s-%s", corev1secret.Name, h)
-		err = populateMap(m, corev1secret, nameWithHash)
+		err = populateMap(m, unstructuredSecret, nameWithHash)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func constructConfigmapAndGenerateName(cm manifest.ConfigMap) (*unstructured.Unstructured, string, error) {
```

Comment:
s/construct/make/

here and below.   because make is shorter and more "go"-ish